v2_migration_guide.md

```diff
@@ -0,0 +1,353 @@
+## Usage with a client
+
+When you have to deal with more than one API keys in your project, or need more flexibility.
+
+```go
+import (
+    "context"
+
+    "github.com/clerk/clerk-sdk-go/v2"
+    "github.com/clerk/clerk-sdk-go/v2/organization"
+)
+
+func main() {
+    ctx := context.Background()
+    config := &organization.ClientConfig{}
+    config.Key = "sk_live_XXX"
+    client := organization.NewClient(config)
+
+    // Create an organization
+    org, err := client.Create(ctx, &organization.CreateParams{
+        Name: clerk.String("Acme Inc"),
+    })
+    if err != nil {
+        if apiErr, ok := err.(*clerk.APIErrorResponse); ok {
+            // Access the API errors and additional information
+            apiErr.TraceID
+            apiErr.Error()
+            apiErr.Response.RawJSON
+        }
+    }
+    // List all organizations, limit results to one.
+    params := &organization.ListParams{}
+    params.Limit = clerk.Int64(1)
+    list, err := organization.List(ctx, params)
+    if err != nil {
+        // handle the error
+    }
+    if list.TotalCount > 0 {
+        // Get the first organization in the list
+        org = list.Organizations[0]
+    }
+}
+```
+
+## List operation responses
+
+A lot of APIs support operations where a resource list is returned. In `v2` of the Clerk Go SDK,
+the return types of list operations are similar. They always contain the total count of resources
+available in the server, while the (sometimes filtered or paginated) results are returned in a slice.
+
+Here's an example. Replace `$resource$` with a resource (package) name.
+
+```go
+import (
+    "github.com/clerk/clerk-sdk-go/v2"
+    "github.com/clerk/clerk-sdk-go/v2/$resource$"
+)
+
+ctx := context.Background()
+list, err := $resource$.List(ctx, &$resource$.ListParams{})
+// If $resource$ was user, the following line would read
+// fmt.Println(list.TotalCount, list.Users)
+fmt.Println(list.TotalCount, list.$resource$s)
+```
+
+## Different types
+
+In the library's `v1` version, some operations would return the same type, while others wouldn't.
+
+The `v2` version moves to a more resource-oriented approach. Each API resource has a type and all types
+are defined in the `clerk` package.
+
+This means there are types for `clerk.User` and `clerk.Domain` and these types are returned by
+the Users and Domains API operations respectively.
+
+All types were consolidated and updated to match the currently supported request parameters and responses
+of the [Clerk Backend API](https://clerk.com/docs/reference/backend-api). Many fields have been renamed and a lot
+of new fields have been added. Deprecated struct fields have been dropped.
+
+## Operation parameters
+
+The `v2` version of Clerk SDK Go introduces another important change about types that can be used as API operation
+parameters. Every field for these structs is a pointer.
+
+The `v2` version of the library introduces helper functions to cast basic type values to pointers. These are:
```

Comment:
❓ Not sure if `cast` is the right verb here 🤔 I see `generates a pointer` here:

https://go.dev/tour/moretypes/1

Reply:
Updated.